autometa/__main__.py

```diff
@@ -0,0 +1,157 @@
+        Overwrite default logging level behavior with provided `level`.
+        This must be a constant from logging levels.
+        See https://docs.python.org/3/library/logging.html#levels for details.
+        i.e. logging.DEBUG, logging.INFO, etc. translates to 0,10, etc...
+
+    Returns
+    -------
+    logging.Logger
+        logging's Logger object to emit messages via methods:
+        'warn','info','debug','error','exception','critical','fatal'
+
+    Raises
+    -------
+    TypeError
+        `level` must be an int
+    ValueError
+        `level` must be one of 0, 10, 20, 30, 40, 50
+    """
+    levels = {
+        logging.NOTSET,
+        logging.DEBUG,
+        logging.INFO,
+        logging.WARNING,
+        logging.ERROR,
+        logging.CRITICAL}
+    if type(level) is not int:
+        raise TypeError(f'{level} must be an int! {type(level)}')
+    if level and level not in levels:
+        raise ValueError(f'{level} not in levels: {levels}!')
+    formatter = logging.Formatter(
+        fmt='[%(asctime)s %(levelname)s] %(name)s: %(message)s',
+        datefmt='%m/%d/%Y %I:%M:%S %p')
+    # Construct file/stream logging handlers
+    streamhandler = logging.StreamHandler()
+    streamhandler.setFormatter(formatter)
+    if fpath:
+        filehandler = logging.FileHandler(fpath)
+        filehandler.setFormatter(formatter)
+        logger.addHandler(filehandler)
+
+    streamhandler.setLevel(level)
+    logger.addHandler(streamhandler)
+    logger.setLevel(logging.DEBUG)
+    return logger
+
+def main(args):
+    # Setup logger
+    # timestamp = time.strftime("%Y-%m-%d_%H-%M-%S",time.gmtime())
+    # log_fpath = args.log if args.log else f'{timestamp}_autometa.log'
+    if args.debug:
+        logger = init_logger(fpath=args.log, level=logging.DEBUG)
+    else:
+        logger = init_logger(fpath=args.log)
+    # Configure AutometaUser
+    # TODO: master from WorkQueue is AutometaUser
+    user = AutometaUser(nproc=args.cpus)
+    user.configure(dryrun=args.check_dependencies)
+
+    for config in args.config:
+        # TODO: Add directions to master from WorkQueue
+        mgargs = user.prepare_binning_args(config)
+        user.run_binning(mgargs)
+        # user.refine_binning()
+        # user.process_binning()
+    # user.get_pangenomes()
+
+def entrypoint():
+    import argparse
+    import time
+    cpus = mp.cpu_count()
+    parser = argparse.ArgumentParser(description='Main script to run Autometa pipeline.')
+    parser.add_argument('config',
+        help='</path/to/metagenome.config>',
+        nargs='*')
+    parser.add_argument('--cpus',
+        help=f'Num. cpus to use when updating/constructing databases (default: {cpus} cpus)',
+        type=int,
+        default=cpus)
+    parser.add_argument('--debug',
+        help='Stream debugging information to terminal',
+        action='store_true',
+        default=False)
+    parser.add_argument('--log', help='</path/to/autometa.log>', type=str)
```

Comment:
As this is a user-facing help message, you should make this more friendly. For example "Path to write a log file". Question that arises from this - should this be set from the config file? It might be nice to have that there as it would help people organize their outputs.

Reply:
We could place this as an additional file in metagenome.config under the `[files]` section

Reply:
This is outside of the scope of this issue and if we think this is necessary, we should add this as a separate issue.

Reply:
Here we still have `help='</path/to/autometa.log>'`, when I think it would be friendlier to write something like "Path to write a log file".